Source/gui/SNTAboutWindowView.swift

```diff
@@ -74,6 +81,120 @@ struct SNTAboutWindowView: View {
   }
 }
 
+// SyncLogReceiver is a helper class to receive logs from the sync service,
+// storing them in an array for later retrieval if the sync fails.
+class SyncLogReceiver: NSObject, SNTSyncServiceLogReceiverXPC {
+  private var logs: [String] = []
+
+  func didReceiveLog(_ log: String) {
+    logs.append(log)
+  }
+
+  func getLogs() -> [String] {
+    return logs
+  }
+
+  func clear() {
+    logs.removeAll()
+  }
+}
+
+struct SyncButtonView: View {
+  @State private var inProgress = false
+  @State private var syncStatus: SNTSyncStatusType = .unknown
+  @State private var lr: MOLXPCConnection?
+
+  let logReceiver = SyncLogReceiver()
+
+  func sync() {
+    logReceiver.clear()
+    inProgress = true
+
+    let ss = SNTXPCSyncServiceInterface.configuredConnection()
+    ss?.invalidationHandler = {
+      DispatchQueue.main.sync {
+        inProgress = false
+        syncStatus = .failedXPCConnection
+      }
+    }
+    ss?.resume()
+
+    let logListener = NSXPCListener.anonymous()
+    lr = MOLXPCConnection(serverWith: logListener)
+    lr?.exportedObject = logReceiver
+    lr?.unprivilegedInterface = NSXPCInterface(with: SNTSyncServiceLogReceiverXPC.self)
+    lr?.resume()
+
+    let proxy = ss?.remoteObjectProxy as? SNTSyncServiceXPC
+    proxy?.sync(withLogListener: logListener.endpoint, syncType: .normal) { status in
+      lr = nil
+
+      DispatchQueue.main.sync {
+        inProgress = false
+        syncStatus = status
+      }
+
+      if status == .success {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
```

Comment:
Minor nit - 5 seconds felt a little long looking at the video, but I don't have a strong opinion if it looked fine to you.